libs/utils/src/properties.c

```diff
@@ -191,324 +481,449 @@ static void parseLine(const char* line, celix_properties_t *props) {
+    celix_properties_entry_t* entry = NULL;
+    if (properties) {
+        entry = celix_stringHashMap_get(properties->map, key);
+    }
+    return entry;
+}
 
-                str = hashMapEntry_getValue(entry);
-                for (int i = 0; i < strlen(str); i += 1) {
-                    if (str[i] == '#' || str[i] == '!' || str[i] == '=' || str[i] == ':') {
-                        fputc('\\', file);
-                    }
-                    fputc(str[i], file);
-                }
+celix_status_t celix_properties_set(celix_properties_t* properties, const char* key, const char* value) {
+    return celix_properties_createAndSetEntry(properties, key, &value, NULL, NULL, NULL, NULL);
+}
 
-                fputc('\n', file);
+celix_status_t celix_properties_setWithoutCopy(celix_properties_t* properties, char* key, char* value) {
+    if (properties) {
+        if (!key || !value) {
+            celix_err_push("Failed to set (without copy) property. Key or value is NULL.");
+            free(key);
+            free(value);
+            return CELIX_ILLEGAL_ARGUMENT;
+        }
+        celix_properties_entry_t* entry = celix_properties_createEntryWithNoCopy(properties, value);
+        if (!entry) {
+            celix_err_push("Failed to create entry for property.");
+            free(key);
+            free(value);
+            return CELIX_ENOMEM;
+        }
 
-            }
-            hashMapIterator_destroy(iterator);
+        bool alreadyExist = celix_stringHashMap_hasKey(properties->map, key);
+        celix_status_t status = celix_stringHashMap_put(properties->map, key, entry);
+        if (status != CELIX_SUCCESS) {
+            celix_err_pushf("Failed to put entry for key %s in map.", key);
+            free(key);
+            celix_properties_destroyEntry(properties, entry);
+        } else if (alreadyExist) {
+            free(key);
         }
-        fclose(file);
-    } else {
-        perror("File is null");
+        return status;
     }
+    return CELIX_SUCCESS; // silently ignore NULL properties, key or value
+}
+
+celix_status_t
+celix_properties_setEntry(celix_properties_t* properties, const char* key, const celix_properties_entry_t* entry) {
+    if (entry) {
+        switch (entry->valueType) {
+        case CELIX_PROPERTIES_VALUE_TYPE_LONG:
+            return celix_properties_setLong(properties, key, entry->typed.longValue);
+        case CELIX_PROPERTIES_VALUE_TYPE_DOUBLE:
+            return celix_properties_setDouble(properties, key, entry->typed.doubleValue);
+        case CELIX_PROPERTIES_VALUE_TYPE_BOOL:
+            return celix_properties_setBool(properties, key, entry->typed.boolValue);
+        case CELIX_PROPERTIES_VALUE_TYPE_VERSION:
+            return celix_properties_setVersion(properties, key, entry->typed.versionValue);
+        default: // STRING
+            return celix_properties_set(properties, key, entry->value);
```

Comment:
This seems unnatural. Our user is not instructed how to construct an entry, and he/she may not know that `entry->typed.strValue` is ignored.

Reply:
I agree, updated this to entry->typed.strValue and updated the code documentation to indicate what is used in the entry for copying.